_posts/2021-5-12-My-OTP-24-Highlights.md

```diff
@@ -0,0 +1,546 @@
+achieve up to 4 times the throughput vs the old implementation. In others,
+there is no difference or even a loss of throughput. So, as always, you need to
+measure and check for yourself!
+
+[gen_tcp]: https://erlang.org/doc/man/gen_tcp.html
+[inet]: https://erlang.org/doc/man/inet.html
+[inet:i/0]: https://erlang.org/doc/man/inet.html#i-0
+[socket]: https://erlang.org/doc/man/socket.html
+[opaque]: http://erlang.org/doc/reference_manual/typespec.html#type-declarations-of-user-defined-types
+
+# EEP-56: Supervisor automatic shutdown #
+
+When creating supervisor hierarchies for applications that manage connections
+such as [ssl] or [ssh], there are times when there is a need for terminating
+that supervisor hierarchy from within. Some event happens on the socket that
+should trigger a graceful shutdown of the processes associated with the
+connection.
+
+Normally this would be done by using [supervisor:terminate_child/2]. However,
+this has two problems.
+
+1. It requires the child to know the ID of the child that needs to be terminated.
+   This is simple when there is just one process in the supervisor, but when there
+   are supervisors under supervisors, this becomes harder and harder to figure out.
+2. Calling [supervisor:terminate_child/2] is a synchronous operation, meaning that
+   if you do the call in the child, you may end up in a deadlock as the top
+   supervisor want to terminate the child while the child is blocking in the call
+   to terminate itself.
+
+To solve this problem [EEP-56] has added a mechanism in which a child can be
+marked as significant and if such a child terminates, it can trigger an automatic
+shutdown of the supervisor that it is part of.
+
+This way a child process can trigger the shutdown of a supervisor hierarchy from
+within, without the child having to know anything about the supervisor hierarchy
+nor risking dead-locking itself during termination.
+
+You can read more about automatic shutdown in the [supervisor documentation].
+
+[ssl]: https://erlang.org/doc/man/ssl.html
+[ssh]: https://erlang.org/doc/man/ssh.html
+[EEP-56]: https://www.erlang.org/erlang-enhancement-proposals/eep-0056.html
+[supervisor:terminate_child/2]: https://erlang.org/doc/man/supervisor.html#terminate_child-2
+[supervisor documentation]: https://erlang.org/doc/man/supervisor.html#auto_shutdown
+
+
+# Edwards-curve Digital Signature Algorithm #
+
+With Erlang/OTP 24 comes support for [Edwards-curve Digital Signature Algorithm]
+(`EdDSA`). `EdDSA` can be used when connecting to or acting as a TLS 1.3
+client/server.
+
+`EdDSA` is a type of [elliptic curve signature algorithm] (`ECDSA`)
+that can be used for secure communication. The security of `ECDSA` relies on a
+[strong cryptographically secure random number] which can cause issues when
+the random number is by mistake not secure enough, as has been the case in several
+uses of ECDSA (none of them in Erlang as far as we know :).
```

Comment:
(none of them in Erlang as far as we know :) )

If `)` is a part of a smily then there is no ending of the paranthesis

Reply:
I intentionally did it like that and I think I'll leave it.